de.metas.business/src/main/java-legacy/org/compiere/model/MInOut.java

```diff
@@ -1306,6 +1306,14 @@ private boolean isCheckCreditLimitNeeded()
 			return false;
 		}
 
+		final IBPartnerStatsDAO bpartnerStatsDAO = Services.get(IBPartnerStatsDAO.class);
+		final I_C_BPartner partner = InterfaceWrapperHelper.create(getCtx(), getC_BPartner_ID(), I_C_BPartner.class, get_TrxName());
+		final BPartnerStats stats = bpartnerStatsDAO.getCreateBPartnerStats(partner);
```

Comment:
introduce IBPartnerStatsDAO.getCreateBPartnerStats(final int bpartnerId).
Load the BP out of transaction.